sawmill-core/src/main/java/io/logz/sawmill/processors/DateProcessor.java

```diff
@@ -69,73 +69,94 @@
-                .toFormatter());
-        dateTimePatternToFormatter.put("UNIX_MS", new DateTimeFormatterBuilder()
-                .appendValue(ChronoField.INSTANT_SECONDS, 1, 10, SignStyle.NEVER)
-                .appendValue(ChronoField.MILLI_OF_SECOND, 3)
-                .toFormatter());
+        dateTimePatternToFormatter.put("ISO8601", ISO8601);
+        dateTimePatternToFormatter.put("UNIX", UNIX);
+        dateTimePatternToFormatter.put("UNIX_MS", UNIX_MS);
+        dateTimePatternToFormatter.put("ELASTIC", ELASTIC);
     }
 
     private final String field;
     private final String targetField;
     private final List<String> formats;
     private final List<DateTimeFormatter> formatters;
     private final ZoneId timeZone;
+    private final DateTimeFormatter outputFormatter;
 
-    public DateProcessor(String field, String targetField, List<String> formats, ZoneId timeZone) {
+    public DateProcessor(String field, String targetField, List<String> formats, ZoneId timeZone, String outputFormat) {
         checkState(CollectionUtils.isNotEmpty(formats), "formats cannot be empty");
         this.field = requireNonNull(field, "field cannot be null");
         this.targetField = requireNonNull(targetField, "target field cannot be null");
         this.formats = formats;
         this.timeZone = timeZone;
 
+        computeFormatter(outputFormat);
+
+        outputFormatter = getDateTimeFormatter(timeZone, outputFormat);
+
         this.formatters = new ArrayList<>();
 
         formats.forEach(format -> {
-            try {
-                dateTimePatternToFormatter.computeIfAbsent(format, k -> new DateTimeFormatterBuilder().parseCaseInsensitive().appendPattern(format).toFormatter());
-            } catch (IllegalArgumentException e) {
-                throw new RuntimeException(String.format("failed to create date processor, format [%s] is not valid", format), e);
-            }
-            DateTimeFormatter formatter = dateTimePatternToFormatter.get(format);
-            if (format.toUpperCase().startsWith("UNIX")) {
-                formatter = formatter.withZone(timeZone == null ? ZoneId.of("UTC") : timeZone);
-            }
+            computeFormatter(format);
+            DateTimeFormatter formatter = getDateTimeFormatter(timeZone, format);
 
             formatters.add(formatter);
         });
+
+    }
+
+    private DateTimeFormatter getDateTimeFormatter(ZoneId timeZone, String format) {
+        DateTimeFormatter formatter = dateTimePatternToFormatter.get(format);
+        if (format.toUpperCase().startsWith("UNIX")) {
+            formatter = formatter.withZone(timeZone == null ? ZoneId.of("UTC") : timeZone);
+        }
+        return formatter;
+    }
+
+    private void computeFormatter(String format) {
+        try {
+            dateTimePatternToFormatter.computeIfAbsent(format, k -> new DateTimeFormatterBuilder().parseCaseInsensitive().appendPattern(format).toFormatter());
```

Comment:
computeIfAbsent returns the DateTimeFormatter
so we can merge this method with getDateTimeFormatter